system/jlib/jstream.cpp

```diff
@@ -282,3 +283,944 @@ IByteInputStream *createInputStream(int handle)
     return new CFileInputStream(handle);
 }
 
+
+//===========================================================================
+
+// This class ensures that data is read in fixed block sizes - NOT a fixed buffer size.
+// This means the buffer size is likely to be bigger than the block size - the class is passed
+// an initial estimate for the potential overlap.
+
+class CBlockedSerialInputStream : public CInterfaceOf<IBufferedSerialInputStream>
+{
+public:
+    CBlockedSerialInputStream(ISerialInputStream * _input, size32_t _blockReadSize, size32_t expectedPeekSize)
```

Comment:
it is not clear (enough) how expectedPeekSize will be used, bettert to keep internal?